src/main/java/teammates/storage/sqlentity/Notification.java

```diff
@@ -0,0 +1,301 @@
+import teammates.common.util.SanitizationHelper;
+
+import jakarta.persistence.Column;
+import jakarta.persistence.Entity;
+import jakarta.persistence.EnumType;
+import jakarta.persistence.Enumerated;
+import jakarta.persistence.Id;
+import jakarta.persistence.Table;
+
+/**
+ * Represents a unique notification in the system.
+ */
+@Entity
+@Table(name = "Notifications")
+public class Notification extends BaseEntity {
+
+    @Id
+    private String notificationId;
+
+    @Column(nullable = false)
+    private Instant startTime;
+
+    @Column(nullable = false)
+    private Instant endTime;
+
+    @Column(nullable = false)
+    @Enumerated(EnumType.ORDINAL)
+    private NotificationStyle style;
+
+    @Column(nullable = false)
+    @Enumerated(EnumType.ORDINAL)
+    private NotificationTargetUser targetUser;
+
+    @Column(nullable = false)
+    private String title;
+
+    @Column(nullable = false)
+    private String message;
+
+    @Column(nullable = false)
+    private boolean shown;
+
+    @CreationTimestamp
+    @Column(nullable = false, updatable = false)
+    private Instant createdAt;
+
+    @UpdateTimestamp
+    @Column
+    private Instant updatedAt;
+
+    /**
+     * Instantiates a new notification, with ID randomly generated and time fields filled automatically.
+     *
+     * @param startTime start time for the notification to be shown to users
+     * @param endTime notifications are hidden from users after endTime
+     * @param style style of the notification (e.g. success, warning, etc.)
+     * @param targetUser student or instructor
+     * @param title title of the notification
+     * @param message message body of the notification
+     */
+    public Notification(Instant startTime, Instant endTime, NotificationStyle style, NotificationTargetUser targetUser,
+                        String title, String message) {
+        this.setStartTime(startTime);
+        this.setEndTime(endTime);
+        this.setStyle(style);
+        this.setTargetUser(targetUser);
+        this.setTitle(title);
+        this.setMessage(message);
+        this.setCreatedAt(Instant.now());
+
+        UUID uuid = UUID.randomUUID();
+        this.notificationId = uuid.toString();
+        this.shown = false;
+    }
+
+    /**
+     * Instantiates a new notification from {@code NotificationBuilder}.
+     */
+    public Notification(NotificationBuilder builder) {
+        this.setStartTime(builder.startTime);
+        this.setEndTime(builder.endTime);
+        this.setStyle(builder.style);
+        this.setTargetUser(builder.targetUser);
+        this.setTitle(builder.title);
+        this.setMessage(builder.message);
+        if (createdAt == null) {
+            this.setCreatedAt(Instant.now());
+        } else {
+            this.setCreatedAt(createdAt);
+        }
+        this.setUpdatedAt(updatedAt);
+        this.notificationId = builder.notificationId;
+        this.shown = builder.shown;
+
+        assert this.createdAt != null;
```

Comment:
this will likely fail. created at is not generated until the the object is inserted into the db.